deployments/puppet/manifests/init.pp

```diff
@@ -1,15 +1,60 @@
 # Main class that installs and configures the agent
 class signalfx_agent (
-    $config,
-    $package_stage = 'final',
-    $repo_base = 'dl.signalfx.com',
-    $config_file_path = '/etc/signalfx/agent.yaml',
-    $version = 'latest') {
+  $config,
+  $package_stage          = 'final',
+  $repo_base              = 'dl.signalfx.com',
+  $config_file_path       = $::osfamily ? {
+    'debian'  => '/etc/signalfx/agent.yaml',
+    'redhat'  => '/etc/signalfx/agent.yaml',
+    'windows' => 'C:\\ProgramData\\SignalFxAgent\\agent.yaml',
+    'default' => '/etc/signalfx/agent.yaml'
+  },
+  $agent_version          = '',
+  $package_version        = '',
+  $installation_directory = 'C:\\Program Files\\SignalFx\\',
+) {
+
+  $service_name = 'signalfx-agent'
 
   if !$config['signalFxAccessToken'] {
     fail("The \$config parameter must contain a 'signalFxAccessToken'")
   }
 
+  if $::osfamily == 'windows' {
+    if $agent_version == '' {
+      fail("The \$agent_version parameter must be set to a valid SignalFx Agent version")
+    }
+
+    $split_config_file_path = $config_file_path.split("\\\\")
+    $config_parent_directory_path = $split_config_file_path[0, - 2].join("\\")
+
+    package { $service_name:
```

Comment:
Is this ok to be before the `win_repo` class below?

Reply:
I think it is ok, because in line 88 we have  `-> Package[$service_name]` that indicates the ordering. So that `win_repo` is applied before `-> Package[$service_name]`

Reply:
<img width="1638" alt="Screen Shot 2019-07-19 at 10 27 37 AM" src="https://user-images.githubusercontent.com/17557526/61542665-0618d480-aa10-11e9-82ab-31922a6ba637.png">

@keitwb The above graph shows the relationship of resources. It's not very legible but you can see `win_repo` being a applied before the package resource.